### DISCUSSION TITLE
ESP32 does not build `samples/hello_world` with `--sysbuild` enabled.

### DISCUSSION THREAD

**Discussion by n1zzo** [zephyrproject-rtos/zephyr#60928] [Q&A] [answered]
After the latest ESP32 refactoring, building `samples/hello_world` with `--sysbuild` enabled gives an error.
The following patch should fix it, there was a leftover dependency to the CONFIG_SOC flag.

```
diff --git a/soc/xtensa/espressif_esp32/esp32s3/loader.c b/soc/xtensa/espressif_esp32/esp32s3/loader.c
index d6530764f9..ab6da243cd 100644
--- a/soc/xtensa/espressif_esp32/esp32s3/loader.c
+++ b/soc/xtensa/espressif_esp32/esp32s3/loader.c
@@ -20,7 +20,7 @@
 #ifdef CONFIG_BOOTLOADER_MCUBOOT

 #define BOOT_LOG_INF(_fmt, ...) \
-    ets_printf("[" CONFIG_SOC "] [INF] " _fmt "\n\r", ##__VA_ARGS__)
+    ets_printf("[" CONFIG_SOC_SERIES "] [INF] " _fmt "\n\r", ##__VA_ARGS__)

 #define HDR_ATTR __attribute__((section(".entry_addr"))) __attribute__((used))
 ```

Let me know if I should open an Issue or a PR.

**Comment by EricNRS** [comment]
~~Looking at main @ 488fd89e2bd64c60354378fdec33b7e65c713d38, `soc/xtensa/espressif_esp32` doesn't exist since it has been replaced by `soc/xtensa/esp32s3/`.  Any chance you have some a repository mismatch?  Try `west update` to get everything back in sync.~~


Update:  I had this backwards due to looking at my repo during a rebase, sorry about that.

**Comment by n1zzo** [comment]
I tried to build from your commit (488fd89) and I still see `soc/xtensa/espressif_esp32`, it shows up on github too:

https://github.com/zephyrproject-rtos/zephyr/tree/488fd89e2bd64c60354378fdec33b7e65c713d38/soc/xtensa/espressif_esp32

Let me know if I should prepare a PR with the above patch.

**Reply by EricNRS** [reply, marked as answer]
Oops, sorry about that.  You are correct, looks like this is a missed change from the ESP32 folder refactor.  Yes, please prepare a PR for it.  I think the Espressif team is automatically notified, but you might want to mention Marek Matej specifically since he did the refactor and he will know what happened and if there are any other things that were missed.

**Comment by n1zzo** [comment]
[PR](https://github.com/zephyrproject-rtos/zephyr/pull/61703) opened!
